### PR TITLE
GH4278: Update Spectre.Console to 0.49.1

### LINKED ISSUE #4278: Update Spectre.Console to  0.49.1

### PR DESCRIPTION
**Pull request by devlead**
* fixes #4278

### CODE REVIEW

**Comment by devlead** [comment]
Found the issue Spectre.Console.Cli no longer allows default commands to have a `--version` parameter, which Cake relied on to have it's own custom version command.

https://github.com/spectreconsole/spectre.console/commit/131b37fff858f8d5dd483ff369bb3d43a07c1e08#diff-d38c9fa48d9c39f8eec58957b2333d13943f2cbe1adee26f30325e527a56f600L26-R41

Will see if I can find away around that, can't use the default Spectre version command as it will produce a version string that will break Cake cli contract. Overriding command settings `ApplicationVersion` and removing cake version feature should probably achieve the same result for end user.

**Comment by devlead** [comment]
As it's by design breaking change in Spectre.Console.Cli, I just added a workaround to opt-out of this behavior for Cake.Tool.

File: src/Cake/Program.cs [hunk 1]
```diff
@@ -63,7 +64,14 @@ public async Task<int> Run(string[] args)
...
 
-            return await app.RunAsync(args);
+            return await app.RunAsync(
+                args.Select(
+                    arg => arg switch {
+                        // Opt-out Spectre.Console.Cli no version parameter for default command
+                        "-v" => "--ver",
+                        "--version" => "--ver",
+                        _ => arg
+                    }));
```
Comment (devlead) [inline src/Cake/Program.cs:74]: This is a terrible workaround, should've probably just addressed if first arg is `-v` / `--version`... but the problem goes deeper.

Spectre.Console.Cli switched from
```diff
-                if (firstArgument.Equals("--version", StringComparison.OrdinalIgnoreCase) ||
-                    firstArgument.Equals("-v", StringComparison.OrdinalIgnoreCase))
```
to
```diff
+        if (args.Contains("-v") || args.Contains("--version"))
+        {
```

default command behavior changed from first to any supplied argument is `--version` / `-v` will print the version and exit the application. even i.e.  `--` between command and remaining args used.

so will have to think about this a bit more. moving PR to WIP.

**Comment by devlead** [comment]
The breaking `--version` change was revered in Spectre.Console 0.49.1

**Comment by devlead** [comment]
Ah would seem remaining arguments has changed in Specre.Console.Cli will investigate, PR still WIP.

**Comment by devlead** [comment]
@cake-build/cake-team this should be ready for review now.

**Review by patriksvensson — APPROVED** [review]
LGTM 👍

**Pull request merged**